x-pack/plugins/index_management/__jest__/client_integration/home/data_streams_project_level_retention.test.ts

```diff
@@ -0,0 +1,161 @@
+    ]);
+
+    const dataStreamForDetailPanel = createDataStreamPayload({
+      name: 'dataStream1',
+      storageSize: '5b',
+      storageSizeBytes: 5,
+      // metering API mock
+      meteringStorageSize: '156kb',
+      meteringStorageSizeBytes: 156000,
+      meteringDocsCount: 10000,
+    });
+
+    setLoadDataStreamsResponse([
+      dataStreamForDetailPanel,
+      createDataStreamPayload({
+        name: 'dataStream2',
+        storageSize: '1kb',
+        storageSizeBytes: 1000,
+        // metering API mock
+        meteringStorageSize: '156kb',
+        meteringStorageSizeBytes: 156000,
+        meteringDocsCount: 10000,
+        lifecycle: {
+          enabled: true,
+          data_retention: '7d',
+          effective_retention: '5d',
+          globalMaxRetention: '20d',
+          retention_determined_by: MAX_DATA_RETENTION,
+        },
+      }),
+    ]);
+
+    setLoadDataStreamResponse(dataStreamForDetailPanel.name, dataStreamForDetailPanel);
+
+    const indexTemplate = fixtures.getTemplate({ name: 'indexTemplate' });
+    setLoadTemplatesResponse({ templates: [indexTemplate], legacyTemplates: [] });
+    setLoadTemplateResponse(indexTemplate.name, indexTemplate);
+
+    notificationService.setup(notificationsServiceMock);
+    testBed = await setup(httpSetup, {
+      history: createMemoryHistory(),
+      services: {
+        notificationService,
+      },
+      config: {
+        enableProjectLevelRetentionChecks: true,
+      },
+    });
+    await act(async () => {
+      testBed.actions.goToDataStreamsList();
+    });
+    testBed.component.update();
+  });
+
+  it('Should show error when retention value is bigger than project level retention', async () => {
+    const { setLoadDataStreamsResponse, setLoadDataStreamResponse } = httpRequestsMockHelpers;
+
+    const ds1 = createDataStreamPayload({
+      name: 'dataStream1',
+      lifecycle: {
+        enabled: true,
+        data_retention: '7d',
+        effective_retention: '5d',
+        retention_determined_by: MAX_DATA_RETENTION,
+        globalMaxRetention: '20d',
+      },
+    });
+
+    setLoadDataStreamsResponse([ds1]);
+    setLoadDataStreamResponse(ds1.name, ds1);
+
+    testBed = await setup(httpSetup, {
+      history: createMemoryHistory(),
+      url: urlServiceMock,
+      config: {
+        enableProjectLevelRetentionChecks: true,
+      },
+    });
+    await act(async () => {
+      testBed.actions.goToDataStreamsList();
+    });
+    testBed.component.update();
+
+    const { actions } = testBed;
+
+    await actions.clickNameAt(0);
+    actions.clickEditDataRetentionButton();
+
+    // set data retention value
+    testBed.form.setInputValue('dataRetentionValue', '25');
+    // Set data retention unit
+    testBed.find('show-filters-button').simulate('click');
+    testBed.find('filter-option-d').simulate('click');
+
+    await act(async () => {
+      testBed.find('saveButton').simulate('click');
```

Comment:
Not entirely sure what was going on under the hood there, but I spent few hours on that and it was the only way I could manage to get the error to bubble up 😬

Reply:
I understand...sometimes I've had similar issues with tests 😅 I was just thinking if the button gets disabled because we already clicked on it once, and not because there is an error... If that's true the test would pass even if we set the value to something lower than the maximum retention period. Would there be some way to check if there is an error message displayed instead of checking the button? 🤔